src/storage/FileResourceStore.ts

```diff
@@ -117,7 +119,7 @@ export class FileResourceStore implements ResourceStore {
    */
   public async getRepresentation(identifier: ResourceIdentifier): Promise<Representation> {
     // Get the file status of the path defined by the request URI mapped to the corresponding filepath.
-    const path = joinPath(this.rootFilepath, this.parseIdentifier(identifier));
+    const path = joinPath(this.rootFilepath, this.resourceMapper.mapUrlToFilePath(identifier));
```

Comment:
Should joining happen inside of the resource mapper?

Reply:
I would say yes, to match the name of the function and be consistent with the reverse function

Reply:
This would actually change a lot in the fileresourcestore, as most functions (except this one here) just get the string path that has already been determined by parseIdentifier somehwere else. But I just remembered that Ieben told me he made a lot of structural adjustments to the store in #72 and I'm seeing in that PR there is already something like parseIdentifier in a ResourceStoreController and many other new things. We might be doing some double work here / causing some merge conflicts? Not sure if/how to proceed.

Reply:
Yes this does seem to be a bit of a conflict 😅 . My first idea would be to look at the changes in that PR that happen to the file store and see which can potentially be copied to ease the situation here. The changes to the file store there seem to be not that quite extensive there so that might be feasible. Merging that PR first is less of an option since the file store changes are only a small part of it and the other changes still need to be looked at, and just in general merging now while you are also working in the store might cause more conflicts (I'm going to worry about the merge issues afterwards myself).

If that is not really feasible, then perhaps we can keep this function as it is now and open an issue to fix it after #72 has been integrated perhaps. It's a bit hard for me to tell without diving into the code myself here.

/cc @RubenVerborgh

Reply:
At the moment, it seems that #72 is a possible direction, and so is this one. I propose that we just continue with the direction we have here in #142. Yes, that will cause us some pains when merging #72, but we can definitely reuse the tests etc. I think it's important that we fully explore the direction we're taking here in #142; in the end, we'll have two directions that we can compare.

Reply:
…and feel free to reuse relevant bits from #72 of course, possibly via git cherry-picking.